tests/unit/nupic/algorithms/knn_classifier_test.py

```diff
@@ -29,6 +29,30 @@
 
 class KNNClassifierTest(unittest.TestCase):
 
+  def testSparsifyVector(self):
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0)
+    inputPattern = np.array([0, 1, 3, 7, 11], dtype=np.int32)
+    
+    # Untouched
+    inputPattern = classifier._sparsifyVector(inputPattern, doWinners=True)
```

Comment:
Sure, but line 46 does alter `inputPattern`, right?

Either way, it's better to use separate variables so there is no confusion or dependency between tests.